File: lua/2021/star1.lua
function file_exists(filename)
    local f = io.open(filename, "rb")
    if f then f:close() end
    return f ~= nil
end

function lines_from(filename)
    if not file_exists(filename) then return {} end
    lines = {}
    for line in io.lines(filename) do
        lines[#lines+1] = line
    end
    return lines
end

function main()
    local input = lines_from("../../input/2021/input1.txt")
    -- print(input[1])
    
    for i=1,#input do
        -- print(input[i])
        input[i] = tonumber(input[i])
    end

    increases = 0
    for i=1,#input-3 do
        prev_window = input[i] + input[i+1] + input[i+2]
        cur_window = input[i+1] + input[i+2] + input[i+3]
        if(cur_window - prev_window > 0) then
            increases = increases + 1
        end
    end

    print(increases)
end

main()

File: javascript/2021/star1.js
const fs = require('fs');

filename = '../../input/2015/input4.txt'

const vowels = [
    'a', 'e', 'i', 'o', 'u'
];

const bad_combos = [
    'ab', 'cd', 'pq', 'xy'
];

function check_line(line) {
    var vowel_cnt = 0;
    var double_check = false;
    var bad_check = false;
    var character = '';

    for(let index = 0; index < line.length; index++) {
        character = line[index];

        // Check if remaining length exists for criteria to be passed
        if(index < line.length - 1){
            if(bad_combos.indexOf(character + line[index+1])>=0) {
                // If this criteria fails, no need to continue loop
                bad_check = true;
                console.log(line + " is bad from combo");
                return false;
            }
            if(character==line[index+1]) {
                double_check = true;
            }
        }
        if(vowels.indexOf(character)>=0) {
            vowel_cnt++;
        }
    }

    if(vowel_cnt >= 3 && double_check && !bad_check){
        console.log(line + " is nice");
        return true;
    } else {
        console.log(line + " is naughty");
        return false;
    }
}

const file = fs.readFileSync(filename, 'utf-8');

const lines = file.split('\n');

var nice_count = 0;

lines.forEach(line => {
    if(check_line(line)) {
        nice_count++;
    }
});
// console.log("estyydmzothggudf is nice: " + check_line('estyydmzothggudf'));
console.log("Nice count is: " + nice_count);